src/ray/object_manager/connection_pool.h

```diff
@@ -91,10 +91,11 @@ class ConnectionPool {
  private:
   /// A container type that maps ClientID to a connection type.
   using SenderMapType =
-      std::unordered_map<ray::ClientID, std::vector<std::shared_ptr<SenderConnection>>>;
+      std::unordered_map<ray::ClientID,
+                         std::unordered_set<std::shared_ptr<SenderConnection>>>;
   using ReceiverMapType =
       std::unordered_map<ray::ClientID,
-                         std::vector<std::shared_ptr<TcpClientConnection>>>;
+                         std::unordered_set<std::shared_ptr<TcpClientConnection>>>;
```

Comment:
why do you think it can improve perf? In most case, vector's perf is better than unordered_set

Reply:
Removing an element from vector will move the rest elements ahead. Search an element of will go though all the elements. If this removing operation is not used frequently or the number is not big, vector could still be better.

Reply:
the maximum number of connections is the number of node-managers.

Reply:
One noob C++ question:
What hash function does unordered_set use here? the address of the pointer?

Reply:
This number is not client number. There is an unordered_map outside (see ReceiverMapType). So the connection number could be the worker number in one node-manager... 

I think it is the address of the raw point.